### DISCUSSION TITLE
.NET Aspire - Resource Group Control

### DISCUSSION THREAD

**Discussion by FullStackChef** [Azure/azure-dev#4467] [Q&A]
I'm wondering if there is a way (other than infra synth -> manual update) to control the name of the resource group that is created. I.e It currently creates "rg-{environmentName}" - however this isn't particularly scalable, we would generally use something along the lines of 
"{prefix}-{teritory}-{environmentName}-rg-{product}"

Apologies if I've missed something basic - but I've spent the last day trying to figure this out without joy.

**Comment by chhanckegrundens** [comment]
The only way I've found to do this, is to initialize environment names in the AppHost project that are more descriptive. So instead of just a "dev" & "test" environment, I'd created "{projectName}-dev" & "{projectName}-test". Since all our Resource Groups are prefixed with "rg-" as standard practice, this is a workable but ugly solution.

There is [this](https://github.com/Azure/azure-dev/issues/4287) open issue, to have the `azd provision` step honor the resource group name when set in the `azure.yaml` file. Using that value currently causes bigger problems with `azd up`, since the provision step uses the default & the deploy step attempts to use the specified one. This also occurs when attempting to set the name using AZURE_RESOURCE_GROUP in the .env file for a given environment.

Not much help I'm afraid, but hopefully there's something here to get you going.

**Reply by FullStackChef** [reply]
I like your workaround. it sounds like a manageable approach, and I've seen uglier - should serve to unblock me in my current workstream thanks
